src/measure_theory/integration.lean

```diff
@@ -782,6 +782,179 @@ begin
+    begin 
+      filter_upwards [h_mono], simp only [mem_set_of_eq], assume a, assume h, induction n with n ih, 
+      {exact le_refl _}, {exact le_trans (h n) ih}
+    end, 
+    congr rfl (funext $ assume n, lintegral_sub (h_meas _) (h_meas _) 
+      (calc
+        (∫⁻ a, f n a) ≤ ∫⁻ a, f 0 a : lintegral_le_lintegral_ae $ h_mono n
+        ... < ⊤ : h_fin)
+        (h_mono n))
+  ... = lintegral (f 0) - (⨅n, ∫⁻ a, f n a) : ennreal.sub_infi.symm 
+
+/-- Known as Fatou's lemma -/
+lemma lintegral_liminf_le {f : ℕ → α → ennreal} (h_meas : ∀n, measurable (f n)):
+  (∫⁻ a, liminf at_top (λ n, f n a)) ≤ liminf at_top (λ n, lintegral (f n)) := 
+calc 
+  (∫⁻ a, liminf at_top (λ n, f n a)) = ∫⁻ a, ⨆n:ℕ, ⨅i≥n, f i a : 
+     congr rfl (funext (assume a, liminf_eq_supr_infi_of_nat))
+  ... = ⨆n:ℕ, ∫⁻ a, ⨅i≥n, f i a : 
+    lintegral_supr 
+    begin 
+      assume n, apply measurable.infi, assume i, by_cases h : i ≥ n, 
+      {convert h_meas i, simp [h]}, 
+      {convert measurable_const, simp [h]}
+    end
+    begin 
+      assume n m hnm a, simp only [le_infi_iff], assume i hi, 
+      refine infi_le_of_le i (infi_le_of_le (le_trans hnm hi) (le_refl _)) 
+    end
+  ... ≤ ⨆n:ℕ, ⨅i≥n, lintegral (f i) : 
+    supr_le_supr $ assume n, le_infi $ 
+      assume i, le_infi $ assume hi, lintegral_le_lintegral _ _ 
+      $ assume a, infi_le_of_le i $ infi_le_of_le hi $ le_refl _
+  ... = liminf at_top (λ n, lintegral (f n)) : liminf_eq_supr_infi_of_nat.symm
+
+lemma limsup_lintegral_le {f : ℕ → α → ennreal} {g : α → ennreal} 
+  (hf_meas : ∀ n, measurable (f n)) (hg_meas : measurable g) 
+  (h_bound : ∀n, ∀ₘa, f n a ≤ g a) (h_fin : lintegral g < ⊤): 
+  limsup at_top (λn, lintegral (f n)) ≤ ∫⁻ a, limsup at_top (λn, f n a) :=
+calc 
+  limsup at_top (λn, lintegral (f n)) = ⨅n:ℕ, ⨆i≥n, lintegral (f i) : 
+    limsup_eq_infi_supr_of_nat 
+  ... ≤ ⨅n:ℕ, ∫⁻ a, ⨆i≥n, f i a : 
+    infi_le_infi $ assume n, supr_le $ assume i, supr_le $ assume hi, 
+    lintegral_le_lintegral _ _ $ assume a, le_supr_of_le i $ le_supr_of_le hi (le_refl _)
+  ... = ∫⁻ a, ⨅n:ℕ, ⨆i≥n, f i a : 
+    (lintegral_infi_ae 
+      (assume n, 
+           @measurable.supr _ _ _ _ _ _ _ _ _ (λ i a, supr (λ (h : i ≥ n), f i a)) 
+      (assume i, measurable.supr_Prop (hf_meas i))) 
+      (assume n, all_ae_of_all $ assume a, 
+       begin 
+         simp only [supr_le_iff], assume i hi, refine le_supr_of_le i _, 
+         rw [supr_pos _], exact le_refl _, exact nat.le_of_succ_le hi    
+       end ) 
+      (lt_of_le_of_lt 
+        (lintegral_le_lintegral_ae 
+        begin 
+          filter_upwards [all_ae_all_iff.2 h_bound], 
+          simp only [supr_le_iff, mem_set_of_eq],
+          assume a ha i hi, exact ha i
+        end ) 
+        h_fin)).symm
+  ... = ∫⁻ a, limsup at_top (λn, f n a) : 
+    lintegral_congr_ae $ all_ae_of_all $ assume a, limsup_eq_infi_supr_of_nat.symm
+
+/-- Dominated convergence theorem for nonnegative functions -/
+lemma tendsto_lintegral_nn
```

Comment:
Actually, I think the name `dominated_convergence_nn` is better and more helpful.